File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    
    let arr = []

    let left =0
    let right = nums.length-1
    let update = nums.length-1

    while(left<=right){

        let leftVal = nums[left] * nums[left]
        let rightVal = nums[right] *nums[right]

        if(leftVal > rightVal){
            arr[update] = leftVal
            left++
            update--

            
        }else{
            arr[update] = rightVal
            right--
            update--
            
        }
    }

    return arr
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:

        # two pointer 
        # find square of each element
        # compare based sorting

        arr = [0] * len(nums)
        left = 0
        right = len(nums)-1
        k = len(nums)-1


        while(left <= right):

            lv = nums[left] * nums[left]
            rv = nums[right] * nums[right]

            # compare based sorting

            if(lv < rv):
                arr[k] = rv
                right-=1
            else:
                arr[k] = lv
                
                left+=1
            k-=1
        return arr
        